### ISSUE TITLE
False positive import-error under pypy3

### ISSUE THREAD

**Issue by RinCat** [pylint-dev/pylint#4126]
### Steps to reproduce
1. Install pypy3.7 form https://www.pypy.org/download.html
2. Create venv
     `XXX/pypy3.7-v7.3.3-linux64/bin/pypy3 -m venv YYY/pypy3`
    `source YYY/pypy3/bin/activate`
3. Install pylint
    `pip install pylint`
4. Create a python file `test.py`
    `import array`
5. Check file
    `pypy3 -m pylint test.py`

### Current behavior
```
************* Module test
test.py:2:0: C0305: Trailing newlines (trailing-newlines)
test.py:1:0: C0114: Missing module docstring (missing-module-docstring)
test.py:1:0: E0401: Unable to import 'array' (import-error)
test.py:1:0: W0611: Unused import array (unused-import)

-------------------------------------
Your code has been rated at -70.00/10
```

### Expected behavior
The `array` module should imported successful.
```
Python 3.7.9 (7e6e2bb30ac5, Nov 18 2020, 10:55:52)
[PyPy 7.3.3-beta0 with GCC 7.3.1 20180303 (Red Hat 7.3.1-5)] on linux
Type "help", "copyright", "credits" or "license" for more information.
And now for something completely different: ``we are a general framework built
on consistent application of adhoc-ness''
>>>> import array
>>>> 
```

### pylint --version output
```
pylint 2.7.0
astroid 2.5
Python 3.7.9 (7e6e2bb30ac5, Nov 18 2020, 10:55:52)
[PyPy 7.3.3-beta0 with GCC 7.3.1 20180303 (Red Hat 7.3.1-5)]
```

This issue does not occur under pylint version 2.6.2.
```
************* Module test
test.py:2:0: C0305: Trailing newlines (trailing-newlines)
test.py:1:0: C0114: Missing module docstring (missing-module-docstring)
test.py:1:0: W0611: Unused import array (unused-import)

-----------------------------------------------------------------------
Your code has been rated at -20.00/10 (previous run: -70.00/10, +50.00)
```

```
pylint 2.6.2
astroid 2.4.2
Python 3.7.9 (7e6e2bb30ac5, Nov 18 2020, 10:55:52)
[PyPy 7.3.3-beta0 with GCC 7.3.1 20180303 (Red Hat 7.3.1-5)]
```

And does not occur under CPython.
```
************* Module test
test.py:2:0: C0305: Trailing newlines (trailing-newlines)
test.py:1:0: C0114: Missing module docstring (missing-module-docstring)
test.py:1:0: W0611: Unused import array (unused-import)

----------------------------------------------------------------------
Your code has been rated at -20.00/10 (previous run: -20.00/10, +0.00)
```
```
pylint 2.7.0
astroid 2.5
Python 3.7.9 (default, Jan 24 2021, 21:08:58) 
[Clang 11.0.0 ]
```
